miro_preprocessor/main.tf

```diff
@@ -55,21 +55,42 @@ module "miro_image_sorter" {
   topic_none_publish_policy          = "${module.none_topic.publish_policy}"
 }
 
-module "miro_copy_s3_asset" {
-  source                         = "miro_copy_s3_asset"
-  topic_miro_copy_s3_asset_arn   = "${module.catalogue_api_topic.arn}"
-  topic_miro_image_to_dynamo_arn = "${module.topic_miro_image_to_dynamo.arn}"
+module "miro_copy_catalogue_derivative" {
+  source                        = "miro_copy_s3_asset"
+  topic_miro_copy_s3_asset_arn  = "${module.catalogue_api_topic.arn}"
+  topic_forward_sns_message_arn = "${module.topic_miro_image_to_dynamo.arn}"
+
+  lambda_error_alarm_arn       = "${local.lambda_error_alarm_arn}"
+  bucket_destination_asset_arn = "${local.bucket_miro_images_public_arn}"
+  bucket_destination_name      = "${local.bucket_miro_images_public_name}"
+  bucket_source_asset_arn      = "${local.bucket_miro_images_sync_arn}"
+  bucket_source_asset_name     = "${local.bucket_miro_images_sync_name}"
+
+  lambda_description     = "Copy catalogue miro derivatives to Loris s3 bucket"
+  lambda_name            = "miro_copy_catalogue_derivative"
+  is_master_asset        = "false"
+  destination_key_prefix = ""
+}
 
-  lambda_error_alarm_arn         = "${local.lambda_error_alarm_arn}"
-  bucket_miro_images_public_arn  = "${local.bucket_miro_images_public_arn}"
-  bucket_miro_images_public_name = "${local.bucket_miro_images_public_name}"
-  bucket_miro_images_sync_arn    = "${local.bucket_miro_images_sync_arn}"
-  bucket_miro_images_sync_name   = "${local.bucket_miro_images_sync_name}"
+module "miro_copy_catalogue_master" {
+  source                       = "miro_copy_s3_asset"
+  topic_miro_copy_s3_asset_arn = "${module.catalogue_api_topic.arn}"
+
+  lambda_error_alarm_arn       = "${local.lambda_error_alarm_arn}"
+  bucket_destination_asset_arn = "${aws_s3_bucket.wellcomecollection-images.arn}"
+  bucket_destination_name      = "${aws_s3_bucket.wellcomecollection-images.id}"
+  bucket_source_asset_arn      = "${local.bucket_miro_images_sync_arn}"
+  bucket_source_asset_name     = "${local.bucket_miro_images_sync_name}"
+
+  destination_key_prefix = "library"
+  lambda_description     = "Copy catalogue miro master assets to private s3 bucket"
+  lambda_name            = "miro_copy_catalogue_master"
+  is_master_asset        = "true"
```

Comment:
It’s a small thing, but I’d rather the fields in these two modules were ordered in such a way to highlight the different settings. Put all the keys that are the same at the top, then put the differing keys in a second block.

Reply:
👍